src/asmcnc/apps/wifi_app/screen_wifi.py

```diff
@@ -503,7 +503,30 @@ def connect_wifi(self):
                 os.system('echo "update_config=1" | sudo tee --append /etc/wpa_supplicant/wpa_supplicant-wlan0.conf')
                 os.system('echo "country="' + self.country.text + '| sudo tee --append /etc/wpa_supplicant/wpa_supplicant-wlan0.conf')
 
-        self.sm.current = 'rebooting'
+        # Flush all the IP addresses from cache
+        os.system('sudo ip addr flush dev wlan0')
+
+        # Reload the updated wpa_supplicant file
+        os.system('sudo wpa_cli -i wlan0 reconfigure')
+
+        # Restart the DHCP service to allocate a new IP address on the new network
+        os.system('sudo systemctl restart dhcpcd')
+
+        def dismiss_wait_popup(dt):
+            if self.set.wifi_available:
+                wait_popup.popup.dismiss()
+                Clock.unschedule(wifi_error_timeout)

```

Comment:
```
def my_callback(dt):
    pass

# call my_callback every 0.5 seconds
event = Clock.schedule_interval(my_callback, 0.5)

# call my_callback in 5 seconds
event2 = Clock.schedule_once(my_callback, 5)

event_trig = Clock.create_trigger(my_callback, 5)
event_trig()

# unschedule using cancel
event.cancel()

# unschedule using Clock.unschedule
Clock.unschedule(event2)

**# unschedule using Clock.unschedule with the callback
# NOT RECOMMENDED
Clock.unschedule(my_callback)**
```

from: https://kivy.org/doc/stable/api-kivy.clock.html

You can check other instances of unscheduling events in the code as well, but generally you want something along the lines of: 
event = Clock.schedule_.....
if self.event: Clock.unschedule(self.event)

so that you check that the event still exists when you unschedule it, otherwise it WILL cause a crash. 

You would probably also benefit from putting a try-except on any popup dismisses, because if the popup gets dismissed (i.e. by a user clicking off of it) before it's dismissed that will also probably cause a crash. 

Finally, it's good practice to make sure any events that may still be running are cleared up on_leave(). 

The wifi_error_timeout might even be better if it's called by the dismiss function after x amount of time has passed, because then that event never gets set up (and doesn't need to be cleared out) if the wifi setting is a success :)

Reply:
All changes have been implemented apart from the last one as discussed. I have tested it on the rig with valid and invalid credentials, as well as disconnecting from network after pressing connect